### PR TITLE
small contribution

### PR DESCRIPTION
**Pull request by b3nk3i**
use FiniteDuration instead of Long for maxDelay
use Topic.URL instead of creating Topic.Name from url for logAction
publish returns MessageID instead of Unit
add a non blocking publishAsync
.. thanks for writing a cool FP book :)

### CODE REVIEW

File: src/main/scala/cr/pulsar/Topic.scala [hunk 1]
```diff
@@ -3,12 +3,13 @@ package cr.pulsar
...
 
-sealed abstract case class Topic(url: String)
+sealed abstract case class Topic(name: Topic.Name, url: Topic.URL)
 
 object Topic {
   import cats.implicits._
 
-  @newtype case class TopicName(value: String)
+  @newtype case class Name(value: String)
+  sealed abstract class URL(val value: String)
```
Comment (gvolpe) [inline src/main/scala/cr/pulsar/Topic.scala:12]: Why not a `@newtype` ?

Comment (b3nk3i) [reply src/main/scala/cr/pulsar/Topic.scala]: ok changing to newtype

File: src/main/scala/cr/pulsar/Topic.scala [hunk 2]
```diff
@@ -9,7 +9,7 @@ object Topic {
   import cats.implicits._
 
   @newtype case class Name(value: String)
-  sealed abstract class URL(val value: String)
+  @newtype class URL(val value: String)
```
Comment (gvolpe) [inline src/main/scala/cr/pulsar/Topic.scala:12]: It has to be a `case class` :)

Comment (b3nk3i) [reply src/main/scala/cr/pulsar/Topic.scala]: :)

**Pull request merged**